pemFioi/quickAlgo/blockly_interface.js

```diff
@@ -713,6 +713,19 @@ function getBlocklyInterface(maxBlocks, subTask) {
          }
       },
 
+      /**
+       * When the blockly code has an invalid block, we call this function to get the error message.
+       * An invalid block is a block that the user has added, then he deleted the sensor that is used by this function.
+       * So the function is deleted from available blocks but still present in users code. So we must send him a message
+       * to explain the user that his code cannot be used because of this block. This function format this message and
+       * also put in "surbrillance" block that is not allowed.
+       * @param block The id of the blockblock that is used but not allowed.
+       */
+      getInvalidBlockMessage: function(block) {
```

Comment:
It's odd to have a function whose purpose is to return a message, while having the side effect of highlighting the block. It would be better to separate those.

If this is meant as a preparation for when it will be possible to highlight multiple blocks, it's still better to separate the highlighting part, from the message part. (Especially as the message is fixed anyway.)

Reply:
For the next commit, I have removed the function getInvalidBlockMessage and just put the code of it inside of initRun.